### PR TITLE
[Merged by Bors] - fix(scripts/{lint_style,mk_all}): limit return value to 125

### PR DESCRIPTION
**Pull request by grunweg**
so the executable's exit code can be used further in shell scripts, such as mathlib's CI.
See https://unix.stackexchange.com/a/418802 for background why a maximum of 125 is safest (on Linux).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
@adomani Since you raised this issue (thanks again for pointing me in this direction!), would you like to take a look?

File: scripts/lint_style.lean
Comment (adomani) [inline scripts/lint_style.lean:31]:
```suggestion
  return min (numberErrorFiles % 128) 125
```
I wonder whether this can be (ever so slightly) more meaningful:
* reduce module 128=2^7 (since 126 and 127 are excluded for other reasons, but 128 appears to be the meaningful number);
* truncate at 125 to avoid 126 and 127.

File: scripts/mk_all.lean
Comment (adomani) [inline scripts/mk_all.lean:73]:
```suggestion
  return min (numberErrorFiles % 128) 125
```
See above for a justification.

**Review by adomani — APPROVED** [review]
This looks great, thanks!

~~Modulo the modulo,~~

maintainer merge

File: scripts/lint_style.lean
Comment (grunweg) [reply scripts/lint_style.lean]: But this would mishandle the case of exactly 128 errors, right? I just pushed a commit allowing 128 explicitly - it is somewhat longer. Do you think returning an exit code of 128 is an important feature? I am not convinced the longer code is worth it, but could be swayed otherwise.

File: scripts/mk_all.lean
Comment (grunweg) [reply scripts/mk_all.lean]: Changed this similarly; let's keep the discussion in the other comment.

File: scripts/lint_style.lean
Comment (adomani) [reply scripts/lint_style.lean]: Sorry, I got this wrong.  We should special-case *actual value* 0 and everything else.

How about something like
```lean
if numberErrorFiles == 0 then return 0 else
  return max 1 (numberErrorFiles % 125)
```

Comment (grunweg) [reply scripts/lint_style.lean]: This makes more sense! Why should we do *modulo* though, instead of saturating? I'd prefer 130 files to return a code of 125 over 2 (or 5), WDYT?

Comment (adomani) [reply scripts/lint_style.lean]: Sure, I do not have a preference.  There might even be some value into specifying that exit code 1 means there were some errors and leave other error codes for potentially other meanings.  Feel free to pick what you prefer: let's simply make sure that exit code 0 is if and only if `numberErrorFiles = 0`!  :smile:

Comment (grunweg) [reply scripts/lint_style.lean]: I'll let a maintainer weigh in on their preference: I have a weak preference for simply reverting the most recent commit; I'm happy to do either of the other options.

Comment (semorrison) [reply scripts/lint_style.lean:31]: Saturation sounds better to me, but we're in the weeds here. :-)

**Comment by semorrison** [comment]
bors d+

**Comment by grunweg** [comment]
It seems that the current solution satisfies everybody - let's land this. Thanks for the reviews and the keen comments!
bors r+

**Pull request closed without merging**
